src/services/reviews.service.js

```diff
@@ -0,0 +1,29 @@
+import Reviews from '../database/models/reviews.model';
+
+async function getReviews(productId) {
+  const reviews = await Reviews.findAll({ where: { productId } });
+  return reviews;
+}
+
+async function addReview(data) {
+  const addedReview = await Reviews.create(data);
+  return addedReview;
+}
+
+async function removeReviewAsUser(userId, reviewId) {
+  const rmReview = await Reviews.destroy({ where: { id: reviewId, userId } });
+  return rmReview;
+}
+
+async function getAverageRating(productId) {
```

Comment:
I think the getAverageRating() function could be optimized by using a database function to calculate the average rating instead of iterating through all reviews and calculating the average rating in the code. This would reduce the amount of data transferred between the database and the code and would also be more efficient imo

Reply:
Applied that 🚀. GREAT!